tests/test_resource.py

```diff
@@ -485,12 +485,14 @@ def test_method(self) -> Response:
         XResource(event)()
 
         mocked_event_api_send.assert_called_once()
-        mocked_event_api_clear.assert_not_called()
+        assert EventAPI().sent_events == []
+        assert EventAPI().pending_events == []
+        assert EventAPI().failed_events == []
```

Comment:
These two tests do not emit any events, so, it is obvious that these lists are empty from the beginning to the very end 😉 

If you really don't want to remove mocks, I have another solution on how to test this logic fully (not beautiful but at least, it confirms the flow):

```python
class TestEventAwareResource:
    def test_initializes_completely_new_event_api_when_building_resource(self) -> None:
        class XResource(EventAwareResource):
            pass

        with patch.object(EventAPI, "clear", autospec=True) as mocked_clear:
            resource = XResource(event)

        assert resource.event_api is EventAPI()
        mocked_clear.assert_called_once_with(resource.event_api)

    def test_emits_all_pending_events_when_request_is_successfully_handled(self) -> None:
        class XResource(EventAwareResource):
            @add_route("/")
            def test_method(self) -> Response:
                return Response({"message": "x"})

        resource = XResource(event)
        with patch.object(resource, "event_api", autospec=True) as mocked_event_api:
            resource()

        mocked_event_api.send.assert_called_once_with()
        mocked_event_api.clear.assert_not_called()
        mocked_event_api.clear_pending.assert_not_called()

    def test_clears_pending_events_when_request_failed_during_handling(self) -> None:
        class XResource(EventAwareResource):
            @add_route("/")
            def test_method(self) -> None:
                raise TypeError

        resource = XResource(event)
        with patch.object(resource, "event_api", autospec=True) as mocked_event_api:
            resource()

        mocked_event_api.send.assert_not_called()
        mocked_event_api.clear.assert_not_called()
        mocked_event_api.clear_pending.assert_called_once_with()
```